lib/src/handler_client.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'base_client.dart';
+import 'request.dart';
+import 'response.dart';
+
+typedef Future<Response> Handler(Request request);
+typedef void CloseHandler();
```

Comment:
Actually if I can target 1.24 then we can get rid of the typedef since there's now a syntax for it. Let me know if its ok to do that.